optional/abstract_optional.py

```diff
@@ -43,3 +43,6 @@ def map(self, func):
     @abstractmethod
     def flat_map(self, func):
         pass
+
+    __bool__ = is_present
```

Comment:
Is this function assignment?  I've never seen it done this way.  Im not sure how I feel about it.